src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -268,30 +267,30 @@ protected function assertValidMappingConfiguration(array $mappingConfig, $object
      */
     protected function detectMetadataDriver($dir, ContainerBuilder $container)
     {
-        // add the closest existing directory as a resource
         $configPath = $this->getMappingResourceConfigDirectory();
-        $resource = $dir.'/'.$configPath;
-        while (!is_dir($resource)) {
-            $resource = dirname($resource);
-        }
+        $extension = $this->getMappingResourceExtension();
 
-        $container->addResource(new FileResource($resource));
+        if (glob($dir.'/'.$configPath.'/*.'.$extension.'.xml')) {
+            $driver = 'xml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.yml')) {
+            $driver = 'yml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.php')) {
+            $driver = 'php';
+        } else {
+            // add the directory itself as a resource
+            $container->fileExists($dir);
 
-        $extension = $this->getMappingResourceExtension();
-        if (($files = glob($dir.'/'.$configPath.'/*.'.$extension.'.xml')) && count($files)) {
-            return 'xml';
-        } elseif (($files = glob($dir.'/'.$configPath.'/*.'.$extension.'.yml')) && count($files)) {
-            return 'yml';
-        } elseif (($files = glob($dir.'/'.$configPath.'/*.'.$extension.'.php')) && count($files)) {
-            return 'php';
+            return is_dir($dir.'/'.$this->getMappingObjectDefaultName()) ? 'annotation' : null;
         }
 
-        // add the directory itself as a resource
-        $container->addResource(new FileResource($dir));
-
-        if (is_dir($dir.'/'.$this->getMappingObjectDefaultName())) {
-            return 'annotation';
+        // add the closest existing directory as a resource
+        $resource = $dir.'/'.$configPath;
+        while (!is_dir($resource)) {
```

Comment:
To me, the code change in this file only ensures that only *one* resource is added. The remaining logic is untouched. I wouldn't change anyelse than the double resource issue here.